src/dependency-manager/src/configs/environment.builder.ts

```diff
@@ -0,0 +1,63 @@
+import fs from 'fs';
+import yaml from 'js-yaml';
+import path from 'path';
+import { BaseEnvironmentConfig } from './environment-config';
+import { EnvironmentSpecV1 } from './v1-spec/environment';
+
+export class EnvironmentBuilder {
+  static async loadFromFile(filepath: string) {
+    // Make sure the file exists
+    let file_contents;
+    try {
+      const data = fs.lstatSync(filepath);
+      if (data.isFile()) {
+        file_contents = fs.readFileSync(filepath, 'utf-8');
+      }
+    // eslint-disable-next-line no-empty
+    } catch { }
+
+    if (!file_contents) {
+      throw new Error(`Missing or invalid file: ${filepath}`);
+    }
+
+    // Try to parse as json
+    if (filepath.endsWith('.json')) {
+      try {
+        const js_obj = JSON.parse(file_contents);
+        return EnvironmentBuilder.parseAndValidate(js_obj, filepath);
```

Comment:
Move down the validate call to the end of the func - like the other loadFromFile method.